### PR TITLE
add compareTo methode to sort MapDtos by date

### PR DESCRIPTION
**Pull request by tobi-spec**

### CODE REVIEW

File: backend/src/main/java/de/tobias/intestinalinspector/api/FoodMapDto.java [hunk 1]
```diff
@@ -13,11 +13,19 @@
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
-public class FoodMapDto {
+public class FoodMapDto implements Comparable<FoodMapDto>{
```
Comment (jamarob) [inline backend/src/main/java/de/tobias/intestinalinspector/api/FoodMapDto.java:16]: Good understanding of collections api 👍 

One might argue that this violates SOC (Separation of concerns) principle. A dtos job is to be a data holder, nothing more.

I would extract the behaviour into an `ComparatorService` that implements `Comparator` and then use `Collections.sort(someCollection, myCustomComparator)`.

See 
https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/util/Comparator.html 
and
https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/util/Collections.html#sort(java.util.List,java.util.Comparator)

Maybe research and refactor some time in the future?

File: backend/src/main/java/de/tobias/intestinalinspector/api/PainMapDto.java [hunk 2]
```diff
@@ -9,11 +9,19 @@
 @NoArgsConstructor
 @AllArgsConstructor
 @Builder
-public class PainMapDto {
+public class PainMapDto implements Comparable<PainMapDto>{
```
Comment (jamarob) [inline backend/src/main/java/de/tobias/intestinalinspector/api/PainMapDto.java:12]: see comment above

File: backend/src/main/java/de/tobias/intestinalinspector/controller/PainController.java [hunk 3]
```diff
@@ -49,6 +50,7 @@ public PainController(PainService painService, DateService dateService) {
         List<PainDto> painListToMap = map(listOfPain);
         Map<String, List<PainDto>> results = dateService.sortPainByDay(painListToMap);
         PainMapsDto painMapsDto = new PainMapsDto();
+
```
Comment (jamarob) [inline backend/src/main/java/de/tobias/intestinalinspector/controller/PainController.java:53]: This is very similar to what FoodController does. There is a chance you could make use of generics to extract the common behaviour into its own service. (-> refactoring ticket?)

File: frontend/.gitignore [hunk 4]
```diff
@@ -4,7 +4,6 @@
 /node_modules
 /.pnp
 .pnp.js
-/package-lock.json
```
Comment (jamarob) [inline frontend/.gitignore:7]: 👍

**Review by RobRobRoberto — APPROVED** [review]
Sehr gut!

File: frontend/src/pages/PerpetratorTracker.js [hunk 5]
```diff
@@ -27,7 +27,7 @@ export default function PerpetratorTracker(){
         let valuesEachDay = 0
 
         pains.forEach(list =>
-            list.map(painDto =>
+            list.forEach(painDto =>
```
Comment (jamarob) [inline frontend/src/pages/PerpetratorTracker.js:30]: Think about refactoring this to make better use of javascrips array methods (I think reduce is what you want here). Due to the imperative style this function is kinda long and not so easy to understand. I think some helper function might improve this.

**Review by jamarob — COMMENTED** [review]
Good Job!👍 Left some nit picky comments for possible refactorings.

**Pull request merged**